File: test_data/code/python/1736-latest-time-by-replacing-hidden-digits.py
class Solution:
    def maximumTime(self, time: str) -> str:
        res = list(time)
        for i, c in enumerate(time):
            if c == '?':
                if i == 0:
                    res[i] = '2'
                elif i == 1:
                    res[i] = '3'
                elif i == 3:
                    res[i] = '5' if any(c == '?' for c in res[0:3]) else '9'
                else:
                    res[i] = '0' if any(c == '?' for c in res[0:2]) else '9'
        return ''.join(res)

File: test_data/code/java/1736-latest-time-by-replacing-hidden-digits.java
class Solution {
    public String maximumTime(String time) {
        char[] arr = time.toCharArray();
        int i = 0;
        if (arr[i] == '?') {
            arr[i] = '2';
        } else {
            i++;
        }
        if (i < 2 && arr[i] == '?') {
            arr[i] = '3';
        } else if (i < 2 && arr[i] != ':') {
            i++;
        }
        if (arr[3] == '?') {
            arr[3] = '5';
        }
        if (arr[4] == '?') {
            arr[4] = '9';
        }
        return String.valueOf(arr);
    }
}